pandas/io/formats/format.py

```diff
@@ -252,8 +252,15 @@ def _get_formatted_index(self):
 
     def _get_formatted_values(self):
         values_to_format = self.tr_series._formatting_values()
+
+        if self.index:
+            leading_space = None
+        else:
+            leading_space = False
```

Comment:
is this really the idiom? These are both False values, so this is extra confusing.

Reply:
shouldn't this just be ``leading_space = self.index``?

Reply:
somehow it looks like there is a slight difference between `False` and `None`, and if default for `leading_space` is `None` which means it's implicit, then a leading_space is added.
```python
if leading_space is False:
     # False specifically, so that the default is
     # to include a space if we get here.
     tpl = u'{v}'
else:
     tpl = u' {v}'
```

Reply:
not sure who added this. It is just ripe for errors. Are there *really* 3 cases or just 2? If there are 3, then the api for leading_space should be a string that is checked and much more readable.

Reply:
should have 2 cases... `leading_space=None` basically is equivalent to True IIUC

Reply:
`leading_space=None` preserves the 0.23.4 and earlier behavior "add a leading space when there are any floating point and no missing values".

Reply:
then let's make this more explicit and use a str argname then like ``leading_space='compat'`` or somesuch

Reply:
@charlesdong1991 can you fix this up